examples/tv-casting-app/linux/CastingShellCommands.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ *
+ *    Copyright (c) 2021 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ * @file Contains shell commands for a ContentApp relating to Content App platform of the Video Player.
+ */
+
+#include "CastingShellCommands.h"
+#include "Casting.h"
+#include <inttypes.h>
+#include <lib/core/CHIPCore.h>
+#include <lib/shell/Commands.h>
+#include <lib/shell/Engine.h>
+#include <lib/shell/commands/Help.h>
+#include <lib/support/CHIPArgParser.hpp>
+#include <lib/support/CHIPMem.h>
+#include <lib/support/CodeUtils.h>
+#include <platform/CHIPDeviceLayer.h>
+
+namespace chip {
+namespace Shell {
+
+static CHIP_ERROR PrintAllCommands()
+{
+    streamer_t * sout = streamer_get();
+    streamer_printf(sout, "  help                 Usage: app <subcommand>\r\n");
+    streamer_printf(sout, "  discover             Discover commissioners. Usage: cast discover\r\n");
+    streamer_printf(sout, "  request <index>      Request commissioning from discovered commissioner with [index]. Usage: cast request 0\r\n");
+    streamer_printf(sout, "  launch <url> <display>   Launch content. Usage: cast launc https://www.yahoo.com Hello\r\n");
+    streamer_printf(sout, "  access <node>        Read and display clusters on each endpoint for <node>. Usage: cast access 0xFFFFFFEFFFFFFFFF\r\n");
+    streamer_printf(sout, "  sendudc <address> <port> Send UDC message to address. Usage: cast sendudc ::1 5543\r\n");
+    streamer_printf(sout, "\r\n");
+
+    return CHIP_NO_ERROR;
+}
+
+static CHIP_ERROR CastingHandler(int argc, char ** argv)
+{
+    CHIP_ERROR error = CHIP_NO_ERROR;
+
+    if (argc == 0 || strcmp(argv[0], "help") == 0)
+    {
+        return PrintAllCommands();
+    }
+    else if (strcmp(argv[0], "discover") == 0)
+    {
+        ChipLogProgress(DeviceLayer, "discover");
+
+        return error = DiscoverCommissioners();
```

Comment:
all these `return error = ....` should be replaced with `return ....`.
Then remove the error variable.

Reply:
done